doc/nrf/libraries/modem/at_parser.rst

```diff
@@ -0,0 +1,103 @@
+.. _at_parser_readme:
+
+AT parser
+#########
+
+.. contents::
+   :local:
+   :depth: 2
+
+Overview
+========
+
+The AT parser is a library that parses AT command responses, notifications, or events.
+It exports an API that lets the user iterate through an AT command string and retrieve any of its elements based on their indices, efficiently and without allocations.
+
+Initialization
+==============
+
+To begin using the AT parser, you need to first initialize it. This is done by invoking the :c:func:`at_parser_init` function, where you provide a pointer to the AT parser and the AT command string that needs to be parsed.
+
+The following code snippet shows how to initialize an AT parser for an AT command response:
+
+.. code-block:: c
+
+   int err;
+   struct at_parser parser = { 0 };
+   const char *at_response = "+CFUN: 1";
+
+   err = at_parser_init(&parser, at_response);
+   if (err) {
+      return err;
+   }
+
+Usage
+=====
+
+Based on the type of the element at a given index, you can obtain its value by calling the type-generic macro :c:macro:`at_parser_num_get` for integer values or the function :c:func:`at_params_string_get` for string values.
+
+The following code snippet shows how to retrieve the prefix, an ``uint16_t`` value, and a string value from an AT command response using the AT parser:
+
+.. code-block:: c
+
+   int err;
+   struct at_parser parser = { 0 };
+   const char *at_response = "+CGCONTRDP: 0,,\"internet\",\"\",\"\",\"10.0.0.1\",\"10.0.0.2\",,,,,1028";
+   uint16_t num;
+   char buffer[16] = { 0 };
+   size_t len = sizeof(buffer);
+
+   err = at_parser_init(&parser, at_response);
+   if (err) {
+      return err;
+   }
+
+   /* Retrieve the AT command response prefix, at index 0 of the AT command string. */
+   err = at_parser_string_get(&parser, 0, buffer, &len);
+   if (err) {
+      return err;
+   }
+
+   /* `at_parser_string_get` does not null-terminate the copied string so null-terminate it. */
+   buffer[len] = '\0';
+
+   /* "Prefix: `+CGCONTRDP`" */
+   printk("Prefix: `%s`\n", buffer);
+
+   /* Retrieve the first subparameter, at index 1 of the AT command string.
```

Comment:
Why is the first parameter (parameter at index 1) a "subparameter"?
What is a "subparameter", in general?

Reply:
`Subparameter` is the terminology used in the `nRF91x1 AT Command Reference Guide` when referring to the values of an AT command string.